tests/_legacy/test_run_settings.py

```diff
@@ -567,3 +569,55 @@ def test_update_env_null_valued(env_vars):
     with pytest.raises(TypeError) as ex:
         rs = RunSettings(sample_exe, run_command=cmd, env_vars=orig_env)
         rs.update_env(env_vars)
+
+
+def test_create_run_settings_run_args_leading_dashes(test_dir, wlmutils):
+    """
+    Test warning for leading `-` in run_args in `exp.create_run_settings`
+    """
+    exp_name = "test-create-run_settings-run_args-leading-dashes"
+    exp = Experiment(exp_name, launcher=wlmutils.get_test_launcher(), exp_path=test_dir)
+
+    run_args = {"--nodes": 1}
+    settings = exp.create_run_settings(
+        "echo", exe_args=["hello", "world"], run_command="srun", run_args=run_args
+    )
+    model = exp.create_model("sr_issue_model", run_settings=settings)
+    exp.start(model)
+
+    statuses = exp.get_status(model)
+    assert all([stat == JobStatus.COMPLETED for stat in statuses])
+
+
+def test_set_run_args_leading_dashes(test_dir, wlmutils):
+    """
+    Test warning for leading `-` for run_args in `settings.set`
+    """
+    exp_name = "test-set-run-args-leading-dashes"
+    exp = Experiment(exp_name, launcher=wlmutils.get_test_launcher(), exp_path=test_dir)
+    settings = exp.create_run_settings(
+        "echo", exe_args=["hello", "world"], run_command="srun"
+    )
+    settings.set("--nodes", "1")
+
+    model = exp.create_model("sr_issue_model", run_settings=settings)
+    exp.start(model)
+    statuses = exp.get_status(model)
+    assert all([stat == JobStatus.COMPLETED for stat in statuses])
+
+
+def test_run_args_integer(test_dir, wlmutils):
+    """
+    Test that `setting.set` will take an integer as a run argument
+    """
+    exp_name = "test-run-args-integer"
+    exp = Experiment(exp_name, launcher=wlmutils.get_test_launcher(), exp_path=test_dir)
+    settings = exp.create_run_settings(
+        "echo", exe_args=["hello", "world"], run_command="srun"
+    )
+    settings.set("--nodes", 1)
+
+    model = exp.create_model("sr_issue_model", run_settings=settings)
+    exp.start(model)
+    statuses = exp.get_status(model)
+    assert all([stat == JobStatus.COMPLETED for stat in statuses])
```

Comment:
Sanity checking: Do we know where these tests came from? It looks like they are using the V0.8 API, so I don't think they came from `smartsim-refactor`. Just wanted to make sure they are not important to the MLI work since they ended up in the `tests/_legacy` dir which is not run in the CI (but I'm guessing they are not, they seem orthogonal at best).

Reply:
Actually, on second thought, these probably came from `develop` as it moved beyond `smartsim-refactor`and `mli-feature`. Should be safe where they are now!